pubsub/integration_test.go

```diff
@@ -1133,8 +1133,8 @@ func TestIntegration_OrderedKeys_Basic(t *testing.T) {
 		})
 		go func() {
 			<-r.ready
-			if r.err != nil {
-				t.Error(r.err)
+			if r.Err != nil {
+				t.Error(r.Err)
```

Comment:
These are breaking changes, which are not allowed in current PubSub version. You shouldn't change field's scope to public, as all the users will get errors in case they are using this field (and they are using, it's implemented to be used by users).

Reply:
Generally speaking, what wat the reason for changing the scope? Was there a way to avoid it?

Reply:
PublishResult has a method Get() that returns the error. At topic_test.go there is a function TestStopPublishOrder that shows the way it can be used, but when I call Get it fails with a timeout. I can't copy TestStopPublishOrder because it uses unexported elements, so I do some kind of mix with TestPSTest function from pstest_test.go file. At docs I found this: "The Get method blocks until a server-generated ID or an error is returned for the published message." I'm trying to make server-generated ID to become returned but I can't find a way to manipulate it [here ](https://cloud.google.com/pubsub/docs/publisher?hl=ru), or anywhere at docs.

Reply:
This would be a good question to ask _a priori_, before making any potentially breaking changes. I'm sure there's a way around it. Another thing to pay attention for is the timeout failure, which could be due to the environment not set up properly, or might also be a bug that can be addressed separately. At any rate, the _primum non nocere_ must not be ignored.

Reply:
I have solved it, please check it